source/vision/providerBase.py

```diff
@@ -7,35 +7,130 @@
 """Module within the vision framework that contains the base vision enhancement provider class.
 """
 
-import driverHandler
-from abc import abstractmethod
-from .constants import Role
+from abc import abstractmethod, ABC
+
+from autoSettingsUtils.autoSettings import AutoSettings
+from baseObject import AutoPropertyObject
 from .visionHandlerExtensionPoints import EventExtensionPoints
-from typing import FrozenSet
+from typing import Optional, Any
 
 
-class VisionEnhancementProvider(driverHandler.Driver):
-	"""A class for vision enhancement providers.
+class VisionEnhancementProviderSettings(AutoSettings, ABC):
+	"""
+	Base class for settings for a vision enhancement provider.
+	Ensure that the following are implemented:
+	- AutoSettings.getId:
+			This is case sensitive. Used in the config file. Does not have to match the module name.
+	- AutoSettings.getTranslatedName:
+			The string that should appear in the GUI as the name
+	- AutoSettings._get_supportedSettings:
+			The "runtime" settings for your provider

```

Comment:
I think that the vision framework introduces some confusion here.

1. Drivers inherrit from AutoSettings. A driver instance is the same as an auto settings instance. Therefore, for AutoSettings, this property returns the settings supported at runtime of the driver.
2. For vision enhancement providers however, an instance of this class is saved on the provider's class. This means that supportedSettings is used to retrieve the settings for the provider, even **before** the provider is actually initialized.

I think this should be clear from the doc string of this class. Therefore, I think the doc string should state that:

* Every vision enhancement provider that implements settings holds a reference to an instance of this class, not the actual class itself.
* For settings that are strictly runtime, the vision enhancement provider should instruct the settings instance at initialization time that it should expose the runtime only settings using this property. exampleProvider_autoGui is a perfect example of this.

Reply:
> This means that `supportedSettings` is used to retrieve the settings for the provider, even before the provider is actually initialized.

Yes, my understanding of the `AutoSettings` method `_get_preInitSettings` is that, as a class method, it relates to the pre-init of the `AutoSettings` class itself, rather than the driver / provider class it might be associated with. Since we have an instance of the settings class, we should use `_get_supportedSettings`. One benefit with this approach is to decouple the settings behaviour from the lifetime of the provider. 

Alternatively, we could choose to communicate the state of the provider to the settings class by calling `_get_preInitSettings` before the provider is initialised, and `_get_supportedSettings`, though I am a bit worried this isn't very explicit.

Another alternative is that we return a class type from `getSettings` on the provider, and we initialise it at the same time as the provider. Though I this adds extra complexity and doesn't really get us anything. 

Also, I note that `_get_preInitSettings` isn't really used anywhere in the code base. We could consider removing it from `AutoSettings`?  Then, in the future, if we wish to update synth / braille drivers to support pre-init settings we can look at what we learn from vision providers.

Reply:
Hmm, I actually think that the code is fine as it is. I think just some extra comments being more explicit about when applies what will solve my concerns.